src/fastpath/fastpath.c

```diff
@@ -472,19 +762,82 @@ void NORETURN fastpath_reply_recv(word_t cptr, word_t msgInfo)
     callerSlot->cteMDBNode = nullMDBNode;
 #endif
 
-    /* I know there's no fault, so straight to the transfer. */
+#ifdef CONFIG_EXCEPTION_FASTPATH
 
+    if (unlikely(fault_type != seL4_Fault_NullFault)) {
+        bool_t restart = 0;
+        switch (fault_type) {
+        case (seL4_Fault_CapFault): {
+            break;
+        }
+        case (seL4_Fault_UnknownSyscall): {
+            break;
+        }
+        case (seL4_Fault_UserException): {
+            break;
+        }
+#ifdef CONFIG_KERNEL_MCS
+        case (seL4_Fault_Timeout): {
+            break;
+        }
+#endif
+
+#ifdef CONFIG_HARDWARE_DEBUG_API
+        case (seL4_Fault_DebugException): {
+            break;
+        }
+#endif
+        default: {
+            restart = 1;
+        }
+        }
+
+        /* TODO: This works here right now because vm faults always restart - the alternative is that the thread becomes inactive,
+         * which would either be handled in the fastpath or redirected into the slowpath (preferred) - either way, this check
+         * will need to be moved above the point of no return to be handled to ensure we don't try and force a switch to an
+         * inactive thread. */
```

Comment:
This does not sound like a TODO you are planning to resolve. Adding an explanation here makes sense (esp on what is implicitly assumed), but it would be good to be slightly clearer that we expect the current implementation to work as is, and that more work will only be needed if we do not have `restart` set.